File: core/src/com/xenoage/zong/commands/core/music/group/BarlineGroupRemove.java
package com.xenoage.zong.commands.core.music.group;

import com.xenoage.utils.document.command.Command;
import com.xenoage.utils.document.command.Undoability;
import com.xenoage.zong.core.music.StavesList;
import com.xenoage.zong.core.music.group.BarlineGroup;


/**
 * Removes a {@link BarlineGroup}.
 * 
 * @author Andreas Wenger
 */
public class BarlineGroupRemove
	implements Command {
	
	//data
	private StavesList stavesList;
	private BarlineGroup group;
	//backup data
	private int groupIndex;
	
	
	public BarlineGroupRemove(StavesList stavesList, BarlineGroup group) {
		this.stavesList = stavesList;
		this.group = group;
	}
	
	
	
	@Override public void execute() {
		//remember element index, since the list is sorted
		groupIndex = stavesList.getBarlineGroups().indexOf(group);
		if (groupIndex == -1)
			throw new IllegalStateException("group is unknown");
		stavesList.getBarlineGroups().remove(group);
	}


	@Override public Undoability getUndoability() {
		return Undoability.Undoable;
	}

	
	@Override public void undo() {
		stavesList.getBarlineGroups().add(groupIndex, group);
	}

}

File: core/src/com/xenoage/zong/commands/core/music/group/BarlineGroupRemove.kt
package com.xenoage.zong.commands.core.music.group

import com.xenoage.utils.document.command.UndoableCommand
import com.xenoage.zong.core.music.StavesList
import com.xenoage.zong.core.music.group.BarlineGroup


/**
 * Removes a [BarlineGroup].
 */
class BarlineGroupRemove(
		private val stavesList: StavesList,
		private val group: BarlineGroup
) : UndoableCommand() {

	//backup
	private var groupIndex: Int = 0

	override fun execute() {
		//remember element index, since the list is sorted
		groupIndex = stavesList.barlineGroups.indexOf(group)
		if (groupIndex == -1)
			throw IllegalStateException("group is unknown")
		stavesList.barlineGroups.remove(group)
	}

	override fun undo() {
		stavesList.barlineGroups.add(groupIndex, group)
	}

}
